pkg/mcp/loader.go

```diff
@@ -198,6 +198,10 @@ func (sm *SessionManager) ensureDeployment(ctx context.Context, server ServerCon
 	}
 
 	if server.Command == "" || !sm.KubernetesEnabled() {
+		if server.URL == "" {
+			return gmcp.ServerConfig{}, fmt.Errorf("MCP server %s needs to update its URL", serverName)
+		}
```

Comment:
We don't have access to the full `MCPServer` object here to check the `.Spec.NeedsURL` field, so I just check for the empty URL, since we set it to empty when there is a URL-hostname conflict.